backend/metrics_backend/model/token_network.py

```diff
@@ -87,10 +95,36 @@ def handle_channel_settled_event(self, channel_identifier: ChannelIdentifier):
         Corresponds to the ChannelSettled event."""
 
         try:
-            self.channels[channel_identifier].update_state(ChannelView.State.SETTLED)
+            channel = self.channels[channel_identifier]
+            channel.update_state(ChannelView.State.SETTLED)
+        
+            self._add_settled_channel_to_participant(channel.participant1)
+            self._add_settled_channel_to_participant(channel.participant2)
         except KeyError:
             log.error(
                 "Received ChannelSettle event for unknown channel '{}'".format(
                     channel_identifier
                 )
             )
+    
+    def get_channel(self, channel_identifier: ChannelIdentifier):
+        if not channel_identifier in self.channels.keys():
+            return None
+        else:
+            return self.channels[channel_identifier]
+
+    def _add_opened_channel_to_participant(self, participant: Address):
+        if not participant in self.participants:
```

Comment:
Why is this tracked here and not in the `MetricsState`?

Reply:
The `MetricsState` counts the values for all networks summed. The `TokenNetwork` for one specific network only. By counting these numbers twice I wanted to avoid to iterate over all the token networks and its participants possibly every 10 seconds.

Reply:
Ah, I missed that.

Reply:
So does that structure make sense to you?

Reply:
I think it's fine, I'd maybe call it `PaymentNetworkMetrics` or something.

And it might make sense to have the same counters for the token network as well, because currently the stuff is calculated every request

Reply:
Or cache the result for a couple of seconds at a time.

Reply:
There is already a TTLCache in use, with a ttl of 10 seconds